lib/osf-components/addon/components/open-resource/component.ts

```diff
@@ -0,0 +1,39 @@
+import { inject as service } from '@ember/service';
+import Component from '@glimmer/component';
+import Intl from 'ember-intl/services/intl';
+
+import { ResourceTypes } from 'ember-osf-web/models/resource';
+import { getBadgeIcon, getBadgeIconDisabled } from 'ember-osf-web/helpers/open-badges-icon-map';
+import Toast from 'ember-toastr/services/toast';
+import RegistrationModel from 'ember-osf-web/models/registration';
+
+interface OpenResourceArgs {
+    registration: RegistrationModel;
+    resourceType: ResourceTypes;
+    hasResource: boolean;
+}
+
+export default class OpenResourceComponent extends Component<OpenResourceArgs> {
+    @service intl!: Intl;
+    @service toast!: Toast;
+
+    constructor(owner: unknown, args: OpenResourceArgs) {
+        super(owner, args);
+    }
+
+    get getSrc() {
+        if (!this.args.resourceType) {
+            return '';
+        }
+
+        try {
+            if (this.args.hasResource === true) {
```

Comment:
Since this component already takes in `@registration` and `@resourceType` as parameters, might as well make `hasResource` a computed property on this component and do something like:
```
get hasResource() {
  if (resourceType === 'data') {
    return this.args.registration.hasData;
  }
 ...
}
```
This way we don't need to pass `@hasResource` as a parameter. Use a switch statement if you'd wish.

Reply:
I agree, however that was not what was requested. I originally had the switch statement and can discuss if you like.

Reply:
You originally had a switch statement for getting the paths to the assets, IIRC. What I am suggesting is to make a new getter named `hasResource` and use a switch/if statement to determine whether the `registration` has the resource based on the `resourceType` argument.